skyportal/handlers/api/photometry.py

```diff
@@ -337,18 +338,104 @@ def post(self):
         df['standardized_flux'] = standardized.flux
         df['standardized_fluxerr'] = standardized.fluxerr
 
+        df["hash"] = df.apply(
+            lambda row: hash(
+                (
+                    row["instrument_id"],
+                    row["filter"],
+                    row["mjd"],
+                    row["standardized_flux"],
+                    row["standardized_fluxerr"],
+                    row["origin"],
+                )
+            ),
+            axis=1,
+        )
+
         instcache = {}
         for iid in df['instrument_id'].unique():
             instrument = Instrument.query.get(int(iid))
             if not instrument:
                 return self.error(f'Invalid instrument ID: {iid}')
             instcache[iid] = instrument
 
+        updated_photometry_groups = False
         for oid in df['obj_id'].unique():
             obj = Obj.query.get(oid)
             if not obj:
                 return self.error(f'Invalid object ID: {oid}')
 
+            # make sure no duplicate data are posted
+            existing_photometry = Obj.get_photometry_owned_by_user(
+                oid, self.current_user
+            )
+            ep = [serialize(phot, "ab", "flux") for phot in existing_photometry]
+            df_existing_photometry = pd.DataFrame(ep)
+            if len(df_existing_photometry) > 0:
+                df_existing_photometry["hash"] = df_existing_photometry.apply(
```

Comment:
Could be worth it to actually store the hash in the DB and index / query on that for finding duplicate phot? In the current solution, you pull all the photometry for an object each time you post new photometry, and cross join all the new photometry against all the existing photometry (without an index), so n^2. If we index the hash, should be O(n).

Reply:
Just for future reference: the reason was inconsistent behavior of `hash`